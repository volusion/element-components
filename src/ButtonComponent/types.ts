import { GlobalSettings } from '../../types';

export interface ButtonConfig {
    buttonStyle: string;
    href: string;
    text: string;
    newWindow: boolean;
}

export interface ButtonProps extends ButtonConfig {
    globalSettings: GlobalSettings;
}

type FontWeight = '300' | '400' | '500' | '600' | '700' | '800' | '900';
export type BorderRadius = 'none' | 'small' | 'medium' | 'large' | 'pill';
export type BorderThickness = 'none' | 'basic' | 'thin' | 'thick';
type TextTransform = 'none' | 'capitalize' | 'lowercase' | 'uppercase';
export type Size = 'small' | 'medium' | 'large' | 'block';
export type Spacing = 'none' | 'tracked' | 'tight' | 'mega';

export interface BaseButtonStylesConfig {
    textColor: string;
    hoverTextColor: string;
    backgroundColor: string;
    hoverBackgroundColor: string;
    borderColor: string;
    hoverBorderColor: string;
    fontWeight: FontWeight;
    textTransform: TextTransform;
    letterSpacing: Spacing;
    borderThickness: BorderThickness;
    size: Size;
    rounded: BorderRadius;
    growOnHover: boolean;
}

export interface VolComponentButton {
    primaryButtonStyles: BaseButtonStylesConfig;
    secondaryButtonStyles: BaseButtonStylesConfig;
}

export type StyleProp = string | number | undefined;

export interface Styles {
    [key: string]: {
        [key: string]: { [key: string]: StyleProp } | StyleProp;
    };
}

export interface BaseButtonProps extends ButtonConfig {
    className?: string;
    globalButtonSettings: BaseButtonStylesConfig;
}
