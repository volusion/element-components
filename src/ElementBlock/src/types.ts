export interface ElementBlockStylesProps {
    isFullWidth: boolean;
    hasHorizontalPadding: boolean;
    topPadding: paddingSizes;
    bottomPadding: paddingSizes;
    showOnDesktop: boolean;
    showOnTablet: boolean;
    showOnMobile: boolean;
}

type paddingSizes = 'None' | 'Small' | 'Medium' | 'Large';

export interface ElementBlockConfigs extends ElementBlockStylesProps {
    customAttrs: customAttrsProps[];
}

export interface ElementBlockProps extends ElementBlockConfigs {
    children: any;
    globalSettings: any;
}

export interface customAttrsProps {
    name: string;
    value: string;
}
export interface ObjectLiteral {
    [key: string]: any;
}
