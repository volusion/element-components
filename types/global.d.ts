// Global definitions for developement

declare module '@volusion/element-proptypes';