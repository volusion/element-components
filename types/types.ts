export interface AphroditeProps {
    StyleSheet: any;
    classes: any;
    css: any;
}
