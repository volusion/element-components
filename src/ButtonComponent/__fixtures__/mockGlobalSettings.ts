import { GlobalSettings } from '../types';
export const mockGlobalSettings: GlobalSettings = {
    globalComponents: {
        volComponentButton: {
            primaryButtonStyles: {
                size: 'large',
                rounded: 'none',
                textColor: '#fff',
                fontWeight: '700',
                borderColor: 'rgba(247, 67, 186, 1)',
                growOnHover: true,
                letterSpacing: 'none',
                textTransform: 'none',
                hoverTextColor: '#fff',
                backgroundColor: 'rgba(247, 68, 186, 1)',
                borderThickness: 'none',
                hoverBorderColor: 'rgba(0, 0, 0, 1)',
                hoverBackgroundColor: 'rgba(218, 15, 148, 1)'
            },
            secondaryButtonStyles: {
                size: 'block',
                rounded: 'none',
                textColor: 'rgba(255, 255, 255, 1)',
                fontWeight: '700',
                borderColor: 'rgba(247, 67, 186, 1)',
                growOnHover: false,
                letterSpacing: 'none',
                textTransform: 'none',
                hoverTextColor: 'rgba(255, 255, 255, 1)',
                backgroundColor: 'rgba(247, 74, 201, 1)',
                borderThickness: 'thin',
                hoverBorderColor: 'rgba(0, 0, 0, 1)',
                hoverBackgroundColor: 'rgba(0, 0, 0, 1)'
            }
        }
    }
};
