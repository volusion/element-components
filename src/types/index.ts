export interface GlobalSettings {
    [key: string]: { [key: string]: any };
}

export interface BlockUtils {
    isAmpRequest: boolean;
}
