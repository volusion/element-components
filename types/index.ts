import { VolComponentButton } from '../src/ButtonComponent/types';

export interface GlobalSettings {
    globalComponents: {
        volComponentButton: VolComponentButton;
    };
}

export interface BlockUtils {
    isAmpRequest: boolean;
}

export interface ComponentProps {
    globalSettings: GlobalSettings;
}
