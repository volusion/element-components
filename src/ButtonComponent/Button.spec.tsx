import React from 'react';
import { mount, render } from 'enzyme';
import { block as ButtonComponent, defaultConfig } from './index';
import { ButtonProps } from './types';
import { mockGlobalSettings as globalSettings } from '../__fixtures__/mockGlobalSettings';

describe('Button component', () => {
    let props: ButtonProps;

    beforeEach(() => {
        props = {
            ...defaultConfig,
            globalSettings
        };
    });

    it('matches snapshot', () => {
        const rendered = render(<ButtonComponent {...props} />);
        expect(rendered).toMatchSnapshot();
    });

    it('uses LinkButton when "href" is passed', () => {
        const href = '#';
        const rendered = mount(<ButtonComponent {...props} href={href} />);
        expect(rendered.find('a').exists()).toBe(true);
    });

    it('uses StandardButton when no "href" is passed', () => {
        const href = '';
        const rendered = mount(<ButtonComponent {...props} href={href} />);
        expect(rendered).toMatchSnapshot();
        expect(rendered.find('a').exists()).toBe(false);
    });
});
