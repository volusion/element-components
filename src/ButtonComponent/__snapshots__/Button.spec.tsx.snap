// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<button
  class="button_1hmnzek"
>
  My Button
</button>
`;

exports[`Button component uses StandardButton when no "href" is passed 1`] = `
<Button
  buttonStyle="primary"
  globalSettings={
    Object {
      "globalComponents": Object {
        "volComponentButton": Object {
          "primaryButtonStyles": Object {
            "backgroundColor": "rgba(247, 68, 186, 1)",
            "borderColor": "rgba(247, 67, 186, 1)",
            "borderThickness": "none",
            "fontWeight": "700",
            "growOnHover": true,
            "hoverBackgroundColor": "rgba(218, 15, 148, 1)",
            "hoverBorderColor": "rgba(0, 0, 0, 1)",
            "hoverTextColor": "#fff",
            "letterSpacing": "none",
            "rounded": "none",
            "size": "large",
            "textColor": "#fff",
            "textTransform": "none",
          },
          "secondaryButtonStyles": Object {
            "backgroundColor": "rgba(247, 74, 201, 1)",
            "borderColor": "rgba(247, 67, 186, 1)",
            "borderThickness": "thin",
            "fontWeight": "700",
            "growOnHover": false,
            "hoverBackgroundColor": "rgba(0, 0, 0, 1)",
            "hoverBorderColor": "rgba(0, 0, 0, 1)",
            "hoverTextColor": "rgba(255, 255, 255, 1)",
            "letterSpacing": "none",
            "rounded": "none",
            "size": "block",
            "textColor": "rgba(255, 255, 255, 1)",
            "textTransform": "none",
          },
        },
      },
    }
  }
  href=""
  newWindow={false}
  text="My Button"
>
  <BaseButton
    buttonStyle="primary"
    globalButtonSettings={
      Object {
        "backgroundColor": "rgba(247, 68, 186, 1)",
        "borderColor": "rgba(247, 67, 186, 1)",
        "borderThickness": "none",
        "fontWeight": "700",
        "growOnHover": true,
        "hoverBackgroundColor": "rgba(218, 15, 148, 1)",
        "hoverBorderColor": "rgba(0, 0, 0, 1)",
        "hoverTextColor": "#fff",
        "letterSpacing": "none",
        "rounded": "none",
        "size": "large",
        "textColor": "#fff",
        "textTransform": "none",
      }
    }
    globalSettings={
      Object {
        "globalComponents": Object {
          "volComponentButton": Object {
            "primaryButtonStyles": Object {
              "backgroundColor": "rgba(247, 68, 186, 1)",
              "borderColor": "rgba(247, 67, 186, 1)",
              "borderThickness": "none",
              "fontWeight": "700",
              "growOnHover": true,
              "hoverBackgroundColor": "rgba(218, 15, 148, 1)",
              "hoverBorderColor": "rgba(0, 0, 0, 1)",
              "hoverTextColor": "#fff",
              "letterSpacing": "none",
              "rounded": "none",
              "size": "large",
              "textColor": "#fff",
              "textTransform": "none",
            },
            "secondaryButtonStyles": Object {
              "backgroundColor": "rgba(247, 74, 201, 1)",
              "borderColor": "rgba(247, 67, 186, 1)",
              "borderThickness": "thin",
              "fontWeight": "700",
              "growOnHover": false,
              "hoverBackgroundColor": "rgba(0, 0, 0, 1)",
              "hoverBorderColor": "rgba(0, 0, 0, 1)",
              "hoverTextColor": "rgba(255, 255, 255, 1)",
              "letterSpacing": "none",
              "rounded": "none",
              "size": "block",
              "textColor": "rgba(255, 255, 255, 1)",
              "textTransform": "none",
            },
          },
        },
      }
    }
    href=""
    newWindow={false}
    text="My Button"
  >
    <Component
      buttonStyle="primary"
      className="button_1hmnzek"
      globalButtonSettings={
        Object {
          "backgroundColor": "rgba(247, 68, 186, 1)",
          "borderColor": "rgba(247, 67, 186, 1)",
          "borderThickness": "none",
          "fontWeight": "700",
          "growOnHover": true,
          "hoverBackgroundColor": "rgba(218, 15, 148, 1)",
          "hoverBorderColor": "rgba(0, 0, 0, 1)",
          "hoverTextColor": "#fff",
          "letterSpacing": "none",
          "rounded": "none",
          "size": "large",
          "textColor": "#fff",
          "textTransform": "none",
        }
      }
      globalSettings={
        Object {
          "globalComponents": Object {
            "volComponentButton": Object {
              "primaryButtonStyles": Object {
                "backgroundColor": "rgba(247, 68, 186, 1)",
                "borderColor": "rgba(247, 67, 186, 1)",
                "borderThickness": "none",
                "fontWeight": "700",
                "growOnHover": true,
                "hoverBackgroundColor": "rgba(218, 15, 148, 1)",
                "hoverBorderColor": "rgba(0, 0, 0, 1)",
                "hoverTextColor": "#fff",
                "letterSpacing": "none",
                "rounded": "none",
                "size": "large",
                "textColor": "#fff",
                "textTransform": "none",
              },
              "secondaryButtonStyles": Object {
                "backgroundColor": "rgba(247, 74, 201, 1)",
                "borderColor": "rgba(247, 67, 186, 1)",
                "borderThickness": "thin",
                "fontWeight": "700",
                "growOnHover": false,
                "hoverBackgroundColor": "rgba(0, 0, 0, 1)",
                "hoverBorderColor": "rgba(0, 0, 0, 1)",
                "hoverTextColor": "rgba(255, 255, 255, 1)",
                "letterSpacing": "none",
                "rounded": "none",
                "size": "block",
                "textColor": "rgba(255, 255, 255, 1)",
                "textTransform": "none",
              },
            },
          },
        }
      }
      href=""
      newWindow={false}
      text="My Button"
    >
      <button
        className="button_1hmnzek"
      >
        My Button
      </button>
    </Component>
  </BaseButton>
</Button>
`;
